lib/install/management/delete.go

```diff
@@ -76,10 +110,82 @@ func (d *Dispatcher) DeleteVCHInstances(conf *metadata.VirtualContainerHostConfi
 		return errors.New(strings.Join(errs, "\n"))
 	}
 
-	if _, err = d.deleteVM(vmm); err != nil {
-		log.Debugf("Error deleting appliance VM %s", err)
+	return nil
+}
+
+func (d *Dispatcher) deleteNetworkDevices(vmm *vm.VirtualMachine, conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Removing appliance VM network devices")
+
+	power, err := vmm.PowerState(d.ctx)
+	if err != nil {
+		log.Errorf("Failed to get vm power status %s: %s", vmm.Reference(), err)
+		return err
+
+	}
+	if power != types.VirtualMachinePowerStatePoweredOff {
+		if _, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+			return vmm.PowerOff(ctx)
+		}); err != nil {
+			log.Errorf("Failed to power off existing appliance for %s", err)
+			return err
+		}
+	}
+
+	spec, err := d.deleteNetworkSpec(vmm, conf)
+	if err != nil {
+		log.Errorf("Unable to create deleting spec: %s", err)
 		return err
 	}
 
+	if spec == nil {
+		log.Infof("No network device attached")
+		return nil
+	}
+	// reconfig
+	info, err := tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return vmm.Reconfigure(ctx, *spec)
+	})
+
+	if err != nil {
+		log.Errorf("Error while removing network devices from appliance VM: %s", err)
+		return err
+	}
+	if info.State != types.TaskInfoStateSuccess {
+		log.Errorf("Removing network devices reported: %s", info.Error.LocalizedMessage)
+		return err
+	}
 	return nil
 }
+
+func (d *Dispatcher) deleteNetworkSpec(vmm *vm.VirtualMachine, conf *metadata.VirtualContainerHostConfigSpec) (*types.VirtualMachineConfigSpec, error) {
```

Comment:
There is a govmomi operation VirtualMachine.RemoveDevice and you can use VirtualDeviceList to get the network cards as it doesn't look like you're using the spec for anything other than device deletion